tests/test_20_url_tools.py

```diff
@@ -0,0 +1,45 @@
+import os
+
+import pytest
+
+from cads_adaptors.tools import url_tools
+
+
+@pytest.mark.parametrize(
+    "urls,expected_nfiles",
+    (
+        (
+            [
+                "https://get.ecmwf.int/repository/test-data/earthkit-data/test-data/test_single.nc"
+            ],
+            1,
+        ),
+        (
+            [
+                "https://get.ecmwf.int/repository/test-data/earthkit-data/test-data/test_single.nc",
+                "https://get.ecmwf.int/repository/test-data/earthkit-data/test-data/test_single.grib",
+            ],
+            2,
+        ),
+    ),
+)
+def test_downloaders(tmp_path, monkeypatch, urls, expected_nfiles):
+    monkeypatch.chdir(tmp_path)  # try_download generates files in the working dir
+    paths = url_tools.try_download(urls)
+    assert len(paths) == expected_nfiles
+
+
+def test_ftp_download(tmp_path, ftpserver):
+    local_test_file = os.path.join(tmp_path, "testfile.txt")
+    with open(local_test_file, "w") as f:
+        f.write("This is a test file")
```

Comment:
`pathlib` replaces `os`, and `tmp_path ` is a `pathlib.Path object`.
```suggestion
    local_test_file = tmp_path / "testfile.txt"
    local_test_file.write_text("This is a test file")
```

Also below you can use methods such as `.read_text`

Reply:
okay, I'm fine with the ol' skool method here